tests/Dockerfile.tests

```diff
@@ -1,32 +1,49 @@
-FROM registry.access.redhat.com/ubi8/go-toolset:1.19 as builder
-COPY . .
+# build vault-manager binary
+FROM registry.access.redhat.com/ubi9/go-toolset:1.22.9 AS builder
+COPY --chown=1001:0 . .
+# Taken from the top-level Makefile
 RUN make gobuild
 
-FROM quay.io/centos/centos:7
-ENV PACKAGES="jq git docker unzip wget"
-ENV BATS_VERSION="v0.4.0"
-ENV VAULT_VERSION="1.9.10"
+# Test image
+# Setup prerequisites for testing
+FROM registry.access.redhat.com/ubi9/ubi:9.5
 
-RUN yum install -y epel-release && \
-    yum install -y $PACKAGES && \
-    rpm -q --queryformat "" $PACKAGES
+# Add Tini
+ENV TINI_VERSION v0.19.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
+
+ENV BATS_VERSION="v1.11.1"
+ENV VAULT_VERSION="1.17.1"
 
 RUN yum install -y \
-      ca-certificates \
-      && \
+    ca-certificates \
+    git \
+    jq \
+    podman \
+    python3-pip \
+    unzip \
+    wget && \
     yum clean all && \
-    update-ca-trust enable && \
     update-ca-trust extract
 
-RUN git clone https://github.com/sstephenson/bats.git && \
-    git --git-dir=bats/.git checkout -b $BATS_VERSION $BATS_VERSION >/dev/null && \
-    bats/install.sh /usr/local
+RUN git clone https://github.com/bats-core/bats-core.git && \
+    git --git-dir=bats-core/.git checkout -b $BATS_VERSION $BATS_VERSION >/dev/null && \
+    bats-core/install.sh /usr/local
+
+RUN wget -q https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip && \
+    unzip vault_${VAULT_VERSION}_linux_amd64.zip && \
+    mv vault /usr/bin
 
-RUN wget https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip && unzip vault_${VAULT_VERSION}_linux_amd64.zip && mv vault /usr/bin
+RUN pip3 install podman-compose
 
 COPY tests/ /tests/
+# copy vault-manager binary from builder
 COPY --from=builder /opt/app-root/src/vault-manager /bin/
 
 WORKDIR /tests
 
-CMD ["bash", "run-tests.sh"]
+ENTRYPOINT ["/tini", "--"]
+
+# Keep the container running by default
+CMD ["sh", "-c", "while true; do sleep 10; done"]
```

Comment:
I don't understand this. Setup -> test suite -> teardown, why need to  use sleep to keep alive?

Reply:
For the compose stack to run the tests, the container has to be available to execute the script.  The `sleep` command was a workaround to keep that image running.  The previous method that used DinD handled the entire lifecycle of each container (which was convenient for running integration tests).   
Efforts were made to implement a more robust solution (NOT using sleep), but the current setup enables tests to run as intended while leveraging podman-compose to orchestrate the stack effectively.
Definitely up for discussing this further.